backend/hdf5/DimensionHDF5.cpp

```diff
@@ -378,6 +378,22 @@ vector<double> RangeDimensionHDF5::ticks() const {
 }
 
 
+vector<double> RangeDimensionHDF5::ticks(size_t start, size_t count) const {
+    vector<double> ticks;
+    ticks.resize(count);
+    H5Group g = redirectGroup();
+    std::string dset_name = g.hasData("ticks") ? "ticks" : "data";
+    DataSet ds = g.openData(dset_name);
+    h5x::DataType memType = data_type_to_h5_memtype(nix::DataType::Double);
+    DataSpace fileSpace, memSpace;
+    nix::NDSize o(1, start);
+    nix::NDSize c(1, count);
+    std::tie(memSpace, fileSpace) = ds.offsetCount2DataSpaces(c, o);
+    ds.read(ticks.data(),  memType, memSpace, fileSpace);
```

Comment:
For out-of-bounds access, I think we would just get a generic `H5Error` exception (from [`DataSet::read`](https://github.com/G-Node/nix/blob/58389c5a9961ee0eb4a7272d0c158ad8ec74e1cb/backend/hdf5/h5x/H5DataSet.cpp#L31)). Maybe `H5Error::value` can tell us this is an out of bounds error and we can convert the exception to an `OutOfBounds` exception. Or, which I think I prefer, we do some manual bounds checking before accessing the data (maybe even in the front-end object) .

Reply:
doing some size checking before rushing into h5 exceptions. Throwing OutOfBounds now, when   size check fails